File: apple/Sources/ReactNativeWatchOS/Modifiers/GridColumnAlignment.swift
import SwiftUI

/// SwiftUI `.gridColumnAlignment(_:)`. Sets the horizontal alignment for the
/// whole `Grid` column the modified cell sits in. Available on watchOS 9+.
enum RNWGridColumnAlignmentModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("gridColumnAlignment") { view, params, _ in
            let alignment = RNWAlignmentParser.horizontal(params.string("alignment"))
            return AnyView(view.gridColumnAlignment(alignment))
        }
    }
}

File: src/modifiers/gridColumnAlignment.ts
import { createModifier } from './createModifier';

/// Horizontal alignment guide for a `Grid` column.
export type GridColumnAlignmentGuide = 'leading' | 'center' | 'trailing';

export interface GridColumnAlignmentParams {
  /// The horizontal alignment to apply to the whole column this cell sits
  /// in.
  alignment: GridColumnAlignmentGuide;
}

/// SwiftUI `.gridColumnAlignment(_:)`. Overrides the horizontal alignment
/// for the entire `Grid` column containing the modified cell.
export function gridColumnAlignment(
  params: GridColumnAlignmentParams
): ReturnType<typeof createModifier>;
export function gridColumnAlignment(
  alignment: GridColumnAlignmentGuide
): ReturnType<typeof createModifier>;
export function gridColumnAlignment(
  a: GridColumnAlignmentParams | GridColumnAlignmentGuide
) {
  if (typeof a === 'string') {
    return createModifier('gridColumnAlignment', { alignment: a });
  }
  return createModifier('gridColumnAlignment', a);
}
